### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by skinny85**
----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by skinny85** [comment]
Second revision after fixing the `update-dependencies` script.

**Comment by skinny85** [comment]
Fixing a NullPointer in the `instal-local-deps.js` script.

**Comment by skinny85** [comment]
Excluded the upgrade of `@node/types` in the update script.

**Comment by RomainMuller** [comment]
Wondering why @dependabot did not bring those up in PR's. I've created an issue on their end to try and understand what's going on.

File: tools/awslint/bin/awslint.ts [hunk 1]
```diff
@@ -265,9 +265,9 @@ function mergeOptions(dest: any, pkg?: any) {
 }
 
 async function shell(command: string) {
-  const child = child_process.spawn(command, undefined, { stdio: [ 'inherit', 'inherit', 'inherit' ]});
+  const child = child_process.spawn(command, [], { stdio: [ 'inherit', 'inherit', 'inherit' ]});
   return new Promise((ok, ko) => {
-    child.once('exit', status => {
+    child.once('exit', (status: any) => {
```
Comment (skinny85) [inline tools/awslint/bin/awslint.ts]: I believe these changes are because of upgrades to `@types/fs-extra` and/or TypeScript.

**Comment by skinny85** [comment]
The build passed 💃🏼

**Comment by skinny85** [comment]
Rebasing.

File: tools/awslint/bin/awslint.ts (hunk 1 above)
Comment (RomainMuller) [reply tools/awslint/bin/awslint.ts]: Might be this code catching a different `@types/node` version due to local dependencies bringing in a `>= 10` or `>= 12` version here... I've had this happen in other places last time.

Comment (skinny85) [reply tools/awslint/bin/awslint.ts]: How did you deal with it?

**Comment by skinny85** [comment]
And again, as the Travis build timed out for some reason.

**Pull request merged**